### DISCUSSION TITLE
Need help to use indicators, how to obtain or convert quotes

### DISCUSSION THREAD

**Discussion by gabrielblaj** [DaveSkender/Stock.Indicators#1016] [Help and support]
//************Python code********************
import ccxt
import pandas as pd
# afiseaza toate ROW
pd.set_option('display.max_rows',None)
import heikin_ashi as ha
from datetime import datetime
import plotly.graph_objects as go# collect the candlestick data from Binance
import schedule
import time
from stock_indicators import indicators
from stock_indicators import Quote

binance = ccxt.binance()
trading_pair = 'BTC/BUSD'
# *****************************limit este numarul de lumanari
candles = binance.fetch_ohlcv(trading_pair, '1h',limit=500)
for candle in candles:
    candle[0] = pd.to_datetime(candle[0],unit='ms')
df=pd.DataFrame(candles)
df.columns= ['date','open', 'high', 'low', 'close', 'volume']
quotes=df 
psar=indicators.get_parabolic_sar(quotes, acceleration_step=0.02, max_acceleration_factor=0.2)
//**********************************end code**********************************************
//******************************error***************************************************
No method matches given arguments for List`1.Add: (<class 'str'>)
//**********************************end errror****************************************

**Comment by DaveSkender** [comment]
> [!TIP]
> For a working example using price quotes from a **pandas.DataFrame**, see our [_Replit_ of the Williams Fractal indicator](https://replit.com/@daveskender/Stock-Indicators-for-Python-Williams-Fractal).  If you're having general trouble getting setup, see our [QuickStart guide](https://github.com/DaveSkender/Stock.Indicators.Python.QuickStart#readme).

See also:

- #1165

**Comment by Rydo111** [comment]
Hi @gabrielblaj; I was getting the same error (`TypeError: No method matches given arguments for List`1.Add: (<class 'str'>)`).

You need to create a list of quotes. The process I'm following:
1. Using your favourite option, get a dataframe containing date, open, high, low, close, volume columns. All are float, except date, which is a datetime.
2. use the code:
`quotes_list = [
        si.Quote(d, o, h, l, c, v)
        for d, o, h, l, c, v
        in zip(df['date'], df['open'], df['high'], df['low'], df['close'], df['volume'], strict=True)
    ]`
to get a list of Quotes
3. pass this list to the stock_indicators function...eg. `indicators.get_zig_zag(quotes_list, EndType.CLOSE, 3)`

I just have to work out [how to convert the answer back to a df](https://github.com/DaveSkender/Stock.Indicators/discussions/1203)! Hope this helps.
